backend/linkbox/linkbox.go

```diff
@@ -0,0 +1,768 @@
+	opt      Options        // options for this backend
+	features *fs.Features   // optional features
+	ci       *fs.ConfigInfo // global config
+}
+
+// Object is a remote object that has been stat'd (so it exists, but is not necessarily open for reading)
+type Object struct {
+	fs          *Fs
+	remote      string
+	size        int64
+	modTime     time.Time
+	contentType string
+	fullURL     string
+	pid         int
+	isDir       bool
+	id          string
+}
+
+// NewFs creates a new Fs object from the name and root. It connects to
+// the host specified in the config file.
+func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, error) {
+	// Parse config into Options struct
+	opt := new(Options)
+	err := configstruct.Set(m, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	ci := fs.GetConfig(ctx)
+
+	f := &Fs{
+		name: name,
+		root: root,
+		opt:  *opt,
+		ci:   ci,
+	}
+
+	f.features = (&fs.Features{
+		CanHaveEmptyDirectories: true,
+		WriteMetadata:           false,
+	}).Fill(ctx, f)
+
+	return f, nil
+}
+
+type Entity struct {
+	Type   string `json:"type"`
+	Name   string `json:"name"`
+	URL    string `json:"url"`
+	Ctime  int64  `json:"ctime"`
+	Size   int    `json:"size"`
+	ID     int    `json:"id"`
+	Pid    int    `json:"pid"`
+	ItemID string `json:"item_id"`
+}
+type Data struct {
+	Entities []Entity `json:"list"`
+}
+type FileSearchRes struct {
+	SearchData Data   `json:"data"`
+	Status     int    `json:"status"`
+	Message    string `json:"msg"`
+}
+
+func getIdByDir(f *Fs, ctx context.Context, dir string) (int, error) {
+	if dir == "" || dir == "/" {
+		return 0, nil // we assume that it is root directory
+	}
+
+	path := strings.TrimPrefix(dir, "/")
+	dirs := strings.Split(path, "/")
+	pid := 0
+
+	for _, tdir := range dirs {
+		pageNumber := 0
+		numberOfEntities := maxEntitiesPerPage
+
+		for numberOfEntities == maxEntitiesPerPage {
+			oldPid := pid
+			pageNumber++
+			requestURL := makeSearchQuery(tdir, pid, f.opt.Token, pageNumber)
+			responseResult := FileSearchRes{}
+			err := getUnmarshaledResponse(ctx, requestURL, &responseResult)
+			numberOfEntities = len(responseResult.SearchData.Entities)
+
+			if err != nil {
+				return 0, err
+			}
+			if len(responseResult.SearchData.Entities) == 0 {
+				return 0, fs.ErrorDirNotFound
+			}
+
+			for _, entity := range responseResult.SearchData.Entities {
+				if entity.Pid == pid && (entity.Type == "dir" || entity.Type == "sdir") && entity.Name == tdir {
+					pid = entity.ID
+				}
+			}
+
+			// This means that we find directory
+			if oldPid != pid {
+				break
+			}
+
+			if pageNumber > 100000 {
+				return 0, fmt.Errorf("too many results")
+			}
+
+		}
+	}
+
+	return pid, nil
+}
+
+func getUnmarshaledResponse(ctx context.Context, url string, result interface{}) error {
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	res, err := http.DefaultClient.Do(req)
```

Comment:
This needs to be using rclone's client in `lib/fshttp.Client`

Reply:
Done